vochain/app.go

```diff
@@ -257,11 +268,23 @@ func (app *BaseApplication) InitChain(_ context.Context,
 // CheckTx unmarshals req.Tx and checks its validity
 func (app *BaseApplication) CheckTx(_ context.Context,
 	req *abcitypes.RequestCheckTx) (*abcitypes.ResponseCheckTx, error) {
+	txReference := vochaintx.TxKey(req.Tx)
+	// store the initial height of the tx
+	initialTTLheight, isReferenced := app.txTTLReferences.LoadOrStore(txReference, app.Height())
+	// check if the tx is referenced by a previous block and the TTL has expired
+	if isReferenced && app.Height() > initialTTLheight.(uint32)+transactionBlocksTTL {
```

Comment:
the `isReferenced` here is redundant, since `app.Height() > initialTTLheight.(uint32)+transactionBlocksTTL` will never be true unless the value was loaded (not stored)
you can simply do

```go
	// retrieve the height when the tx was first checked, or else store the current height
	initialTTLheight, _ := app.txTTLReferences.LoadOrStore(txReference, app.Height())
	// check if the tx is referenced by a previous block and the TTL has expired
	if app.Height() > initialTTLheight.(uint32)+transactionBlocksTTL {
		// remove tx reference and return checkTx error
```